timemachine/fe/free_energy.py

```diff
@@ -1202,8 +1253,13 @@ def replica_from_samples(traj: Trajectory) -> CoordsVelBox:
         if len(initial_states) == 2:
             fraction_accepted_by_pair = fraction_accepted_by_pair[1:]  # remove stats for identity move
 
-        for samples, samples_iter in zip(samples_by_state, samples_by_state_iter):
-            samples.extend(samples_iter)
```

Comment:
nit: should we remove the `Trajectory.extend` method, since it's not used anymore?

Reply:
Seems reasonable to keep as it is a handy utility function, if you want to write some quick and dirty code where you extend a trajectory from the results of calling `sample(initial_state, md_params)`. Can pull out if we want though, as it is no longer used

Reply:
I'm OK either way. In the future we might be able to generalize the `Trajectory` type to handle both in-memory and disk-backed arrays and go back to using `extend`.